ouroboros-consensus/src/Ouroboros/Consensus/Mempool.hs

```diff
@@ -1,6 +1,13 @@
 -- | The mempool API and implementation.
-module Ouroboros.Consensus.Mempool (module X) where
+module Ouroboros.Consensus.Mempool (
+    module Ouroboros.Consensus.Mempool.API
+  , module Ouroboros.Consensus.Mempool.Capacity
+  , module Ouroboros.Consensus.Mempool.Impl
+  , module Ouroboros.Consensus.Mempool.Init
+  ) where
 
-import           Ouroboros.Consensus.Mempool.API as X
-import           Ouroboros.Consensus.Mempool.Impl as X
-import           Ouroboros.Consensus.Mempool.TxSeq as X (TicketNo)
+import           Ouroboros.Consensus.Mempool.API
```

Comment:
Question: There seems to be some coding convention change here, from 
```
module Foo(module X) where

import Bar as X
import Baz as X
```
to
```
module Foo(
  module Bar, 
  module Baz
) where

import Bar
import Baz
```
I don't personally have a strong preference for either but perhaps we should document those conventions somewhere?

Reply:
https://haskell-haddock.readthedocs.io/en/latest/markup.html#re-exporting-an-entire-module
Unless we make the active decision to re-export the whole module (which adds nice haddock hyperlinking to the module), I would suggest doing the re-exports explicitly and not using `module` export. In part because the generated haddock export list will be just a bare list without any sectioning, and this prevents accidental leakage if we export a new definition from the upstream module, in which case its probably better to be explicit

Reply:
https://hackage.haskell.org/package/containers-0.6.7/docs/Data-Map.html is probably a nice example where whole-module re-exports are useful